packages/ffe-file-upload-react/src/FileUpload.js

```diff
@@ -29,97 +33,107 @@ class FileUpload extends React.Component {
     }
 
     render() {
         const {
             id,
             label,
-            selectedFilesHeaderLabel,
-            selectedFiles,
-            accept,
+            files,
+            cancelText,
+            deleteText,
             multiple,
-            errorMessage,
-            infoMessage,
-            successMessage,
+            infoTitle,
+            infoText,
+            infoSubText,
+            uploadTitle,
+            uploadMicroText,
+            uploadSubText,
+            onFileDeleted,
         } = this.props;
 
         return (
             <div className="ffe-file-upload">
-                <button
-                    className="ffe-file-upload__button"
-                    aria-invalid={String(!!errorMessage)}
-                    onClick={this.triggerUploadFileNativeHandler}
+                <div className="ffe-file-upload__info-section">
+                    <div className="ffe-file-upload__info-section__title">
+                        {infoTitle}
+                    </div>
+                    {Object.keys(files).length > 0 ? (
+                        <div className="ffe-file-upload__info-section__selected-files">
+                            {Object.keys(files).map((file, index) => (
+                                <FileItem
+                                    key={index}
+                                    index={index}
+                                    file={files[file]}
+                                    cancelText={cancelText}
+                                    deleteText={deleteText}
+                                    onFileDeleted={onFileDeleted}
+                                />
+                            ))}
+                        </div>
+                    ) : (
+                        <>
+                            <div className="ffe-file-upload__info-section__text">
+                                {infoText}
+                            </div>
+                            <div className="ffe-file-upload__info-section__subtext">
+                                {infoSubText}
+                            </div>
+                        </>
+                    )}
+                </div>
+                <div
+                    className="ffe-file-upload__upload-section"
+                    role="presentation"
+                    onDrop={this.onFilesDropped}
+                    onDragOver={event => {
+                        event.preventDefault();
+                        this.setState({ hover: true });
+                    }}
+                    onDragLeave={() => this.setState({ hover: false })}
                 >
-                    <span className="ffe-file-upload__button__label-icon" />
-                    <span id={`${id}-label`}>{label}</span>
-                </button>
+                    <div
+                        className={`ffe-file-upload__upload-section__border${
+                            this.state.hover
```

Comment:
Rather than toggling the class, styling should be applied to the `:hover` state in CSS/Less.

Reply:
This won't work, as it's only supposed to trigger styling when dragging files over the object, just mousing should not trigger it